command-builder/deploy/index.ts

```diff
@@ -1,31 +1,30 @@
 import { BuilderContext, BuilderOutput, createBuilder } from '@angular-devkit/architect';
 import { JsonObject } from '@angular-devkit/core';
-const NetlifyAPI = require('netlify');
-
+import { Schema } from './schema';
 
-interface Options extends JsonObject {
-  outputPath: string;
-  netlifyToken: string,
-  siteId: string,
-  configuration: string;
-}
+const NetlifyAPI = require('netlify');
 
-export default createBuilder<Options>(
-  async (builderConfig: Options, context: BuilderContext): Promise<BuilderOutput> => {
+export default createBuilder<any>(
+  async (builderConfig: Schema, context: BuilderContext): Promise<BuilderOutput> => {
     context.reportStatus(`Executing deploy...`);
     context.logger.info(`Executing deploy command ...... `);
+    let buildResult;
+    if (builderConfig.noBuild) {
+      context.logger.info(`📦 Skipping build`);
+      buildResult = true;
+    } else {
+      const configuration = builderConfig.configuration ? builderConfig.configuration : 'production';
 
-    const configuration = builderConfig.configuration ? builderConfig.configuration : 'production';
-
-    const build = await context.scheduleTarget({
-      target: 'build',
-      project: context.target !== undefined ? context.target.project : '',
-      configuration
-    });
+      const build = await context.scheduleTarget({
+        target: 'build',
+        project: context.target !== undefined ? context.target.project : '',
```

Comment:
I would throw instead, you need a project – it won't work otherwise.

@mgechev 's code had this:
```

    if (!context.target) {
      throw new Error('Cannot deploy the application without a target');
    }
```

Reply:
Done.